File: HTTP_Server_CLI/scenarios/1.2.2-dynamic_10.lua
-- @label: 10 dynamic routes
-- @group: Dynamic routes
-- @competitors: all
-- Round-robins through all 10 registered dynamic routes.

local counter = 0

request = function()
   counter = counter + 1
   local mod = counter % 10

   if mod == 0 then
      return wrk.format("GET", "/user/" .. (counter % 100))
   elseif mod == 1 then
      return wrk.format("GET", "/post/article-" .. (counter % 100))
   elseif mod == 2 then
      return wrk.format("GET", "/api/v1/resource-" .. (counter % 100))
   elseif mod == 3 then
      return wrk.format("GET", "/category/cat-" .. (counter % 100))
   elseif mod == 4 then
      return wrk.format("GET", "/tag/tag-" .. (counter % 100))
   elseif mod == 5 then
      return wrk.format("GET", "/product/sku-" .. (counter % 100))
   elseif mod == 6 then
      return wrk.format("GET", "/order/ord-" .. (counter % 100))
   elseif mod == 7 then
      return wrk.format("GET", "/invoice/inv-" .. (counter % 100))
   elseif mod == 8 then
      return wrk.format("GET", "/review/rev-" .. (counter % 100))
   else
      return wrk.format("GET", "/comment/cmt-" .. (counter % 100))
   end
end

File: HTTP_Server_CLI/scenarios/php/1.2.2-dynamic_10.php
<?php
// @label: 10 dynamic routes
// @group: Dynamic routes
// @competitors: all
// Round-robins through all 10 registered dynamic routes.

$paths = [];
for ($i = 0; $i < 1000; $i++) {
   $mod = $i % 10;
   $id  = $i % 100;

   $paths[] = match ($mod) {
      0 => "/user/{$id}",
      1 => "/post/article-{$id}",
      2 => "/api/v1/resource-{$id}",
      3 => "/category/cat-{$id}",
      4 => "/tag/tag-{$id}",
      5 => "/product/sku-{$id}",
      6 => "/order/ord-{$id}",
      7 => "/invoice/inv-{$id}",
      8 => "/review/rev-{$id}",
      9 => "/comment/cmt-{$id}",
   };
}

return [
   'method' => 'GET',
   'paths'  => $paths,
];
